src/bun.js/bindings/bindings.cpp

```diff
@@ -458,46 +465,44 @@ bool Bun__deepEquals(JSC__JSGlobalObject* globalObject, JSValue v1, JSValue v2,
     bool v2Array = isArray(globalObject, v2);
     RETURN_IF_EXCEPTION(*scope, false);
 
-    JSObject* o1 = v1.getObject();
-    JSObject* o2 = v2.getObject();
-
     if (v1Array != v2Array)
         return false;
 
     if (v1Array && v2Array) {
         JSC::JSArray* array1 = JSC::jsCast<JSC::JSArray*>(v1);
         JSC::JSArray* array2 = JSC::jsCast<JSC::JSArray*>(v2);
+
         size_t length = array1->length();
         if (length != array2->length()) {
             return false;
         }
 
-        if (array1->canDoFastIndexedAccess() && array2->canDoFastIndexedAccess()) {
```

Comment:
`canDoFastIndexedAccess` checks for holes?

Reply:
From my testing it returns true for both. But we can use `canGetIndexQuickly` inside the loop